.rubocop.yml

```diff
@@ -1,7 +1,9 @@
+---
+require: rubocop-performance
 AllCops:
-  TargetRubyVersion: 2.1
```

Comment:
awsecrets still support Ruby 2.1 🙏

Reply:
Ouch... that we will need to decide which versions of Rubocop + rubocop-performance we want to support.
It seems that for latest version of rubocop (0.79.0) requires Ruby > = 2.3.0.
Take a look here for more on that:
https://travis-ci.org/k1LoW/awsecrets/jobs/645730875?utm_medium=notification&utm_source=github_status
Also, the first time that I installed rubocop-performance (when it was missing), I got some complain about having to upgrade Rubocop as well.

Reply:
Did a quick test here installing Ruby 2.1 with rvm:

```
alceu@yggdrasil:~/Projetos/awsecrets$ ruby -v
ruby 2.1.10p492 (2016-04-01 revision 54464) [x86_64-linux]
alceu@yggdrasil:~/Projetos/awsecrets$ gem install bundler
Fetching bundler-2.1.4.gem
ERROR:  Error installing bundler:
	The last version of bundler (>= 0) to support your Ruby & RubyGems was 1.17.3. Try installing it with `gem install bundler -v 1.17.3`
	bundler requires Ruby version >= 2.3.0. The current ruby version is 2.1.10.492.
alceu@yggdrasil:~/Projetos/awsecrets$ gem install bundler -v 1.17.3
Fetching bundler-1.17.3.gem
Successfully installed bundler-1.17.3
Parsing documentation for bundler-1.17.3
Installing ri documentation for bundler-1.17.3
Done installing documentation for bundler after 5 seconds
1 gem installed
alceu@yggdrasil:~/Projetos/awsecrets$ which bundle
/home/alceu/.rvm/gems/ruby-2.1.10/bin/bundle
alceu@yggdrasil:~/Projetos/awsecrets$ bundle --version
Bundler version 1.17.3
alceu@yggdrasil:~/Projetos/awsecrets$ bundle install --path=.vendor
Fetching gem metadata from https://rubygems.org/.........
public_suffix-4.0.3 requires ruby version >= 2.3, which is incompatible with the current version, ruby 2.1.10p492
```

Before running `bundle install`, I copied `awsecrets.gemspec` the from the master branch:

```
lib = File.expand_path('lib', __dir__)
$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
require_relative 'lib/awsecrets/version'

Gem::Specification.new do |spec|
  spec.name          = 'awsecrets'
  spec.version       = Awsecrets::VERSION
  spec.authors       = ['k1LoW']
  spec.email         = ['k1lowxb@gmail.com']

  spec.summary       = 'AWS credentials loader'
  spec.description   = 'AWS credentials loader'
  spec.homepage      = 'https://github.com/k1LoW/awsecrets'
  spec.license       = 'MIT'

  spec.files         = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
  spec.bindir        = 'exe'
  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
  spec.require_paths = ['lib']

  spec.add_runtime_dependency 'aws-sdk', '>= 2', '< 4'
  spec.add_runtime_dependency 'aws_config', '~> 0.1.0'
  spec.add_development_dependency 'bundler', '>= 1.9', '< 3.0'
  spec.add_development_dependency 'octorelease'
  spec.add_development_dependency 'pry'
  spec.add_development_dependency 'rake', '~> 10.0'
  spec.add_development_dependency 'rspec'
  spec.add_development_dependency 'rubocop', '0.57'
end
```

Just did a merge too with the master branch as well and the result is the same.
Not sure if I'm doing something wrong or the current configuration of Rubocop will just not work.

Reply:
Travis CI master branch build is passing.

https://travis-ci.org/k1LoW/awsecrets/builds/645523693

Can you see the difference between your code and the code of master?

Reply:
These are the differences that I see relevant:

```
diff --git a/.rubocop.yml b/.rubocop.yml
index 8fd7727..3fd58d6 100644
--- a/.rubocop.yml
+++ b/.rubocop.yml
@@ -1,7 +1,9 @@
+---
+require: rubocop-performance
 AllCops:
-  TargetRubyVersion: 2.1
+  TargetRubyVersion: 2.3.4
 
-Lint/HandleExceptions:
+Lint/SuppressedException:
   Enabled: false
 
 Lint/MissingCopEnableDirective:
@@ -69,3 +71,24 @@ Style/SymbolProc:
 
 Style/BracesAroundHashParameters:
   Enabled: false
+
+Style/FrozenStringLiteralComment:
+  Enabled: false
+
+Style/ExpandPathArguments:
+  Enabled: false
+
+Naming/MemoizedInstanceVariableName:
+  Enabled: false
+
+Layout/EmptyLineAfterGuardClause:
+  Enabled: false
+
+Gemspec/OrderedDependencies:
+  Enabled: false
+
+Layout/SpaceAroundOperators:
+  Enabled: false
+
+Style/SafeNavigation:
+  Enabled: false
```
And

```
diff --git a/awsecrets.gemspec b/awsecrets.gemspec
index 6446e6f..a2d4297 100644
--- a/awsecrets.gemspec
+++ b/awsecrets.gemspec
@@ -25,5 +25,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'pry'
   spec.add_development_dependency 'rake', '~> 10.0'
   spec.add_development_dependency 'rspec'
-  spec.add_development_dependency 'rubocop', '0.57'
+  spec.add_development_dependency 'rubocop', '0.79.0'
+  spec.add_development_dependency 'rubocop-performance'
 end
```

Reply:
After reverting those changes:

```
$ bundle exec rake
/usr/bin/ruby2.5 -I/home/alceu/Projects/awsecrets/.vendor/ruby/2.5.0/gems/rspec-core-3.9.1/lib:/home/alceu/Projects/awsecrets/.vendor/ruby/2.5.0/gems/rspec-support-3.9.2/lib /home/alceu/Projects/awsecrets/.vendor/ruby/2.5.0/gems/rspec-core-3.9.1/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb

Awsecrets
  has a version number
  Configuration
    load --profile option
    load --region option
    load secrets.yml
    load secrets_other.yml via AWS_SECRETS_PATH
    load secrets_with_session_token.yml
    load AWS_PROIFLE
    load AWS_DEFAULT_PROIFLE
    load AWS_REGION
    load AWS_DEFAULT_REGION
    load AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
    load AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN
    AssumeRole
      load AWS_PROIFLE=assumed_no_session_name
      load secrets_with_role_arn.yml
      when passing --profile assumed
        should create the proper config
        should call Aws::AssumeRoleCredentials.new with the correct client
        should call Aws::AssumeRoleCredentials.new with the correct role_arn
        should call Aws::AssumeRoleCredentials.new with the correct role_session_name
        should call Aws::AssumeRoleCredentials.new with the correct external_id
      load AWS_PROFILE=assumed
        should create the proper config
        should call Aws::AssumeRoleCredentials.new with the correct client
        should call Aws::AssumeRoleCredentials.new with the correct role_arn
        should call Aws::AssumeRoleCredentials.new with the correct role_session_name
        should call Aws::AssumeRoleCredentials.new with the correct external_id

Awsecrets
  Disable load YAML
    secrets_path: false
  Disable load YAML
    disable_load_secrets: true

Awsecrets
  Precedence
    --profile option > AWS_PROFILE
    --profile option > AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
    --profile option > secrets.yml (No .aws/config)
    --profile option > secrets.yml
    --region option > AWS_REGION
    AWS_REGION > AWS_DEFAULT_REGION

Finished in 0.08395 seconds (files took 0.53691 seconds to load)
32 examples, 0 failures

Running RuboCop...
.rubocop.yml: Style/MultilineOperationIndentation has the wrong namespace - should be Layout
.rubocop.yml: Style/PredicateName has the wrong namespace - should be Naming
Inspecting 11 files
...........

11 files inspected, no offenses detected
```
I guess that installing and requiring `rubocop-performance` brought this mess to the tests (I started inserting those new asserts after installing and requiring it)